leap/leap.go

```diff
@@ -0,0 +1,19 @@
+package main
+
+import "fmt"
+
+func main() {
+	fmt.Println("The program determines if a year is a leap year.")
+	fmt.Println("Enter the year:")
+	var year int
+	fmt.Scan(&year)
+	if year%400 == 0 {
+		fmt.Println(year, " is a leap year.")
```

Comment:
nit: `fmt.Println` adds once space between its arguments. So there is no need to put it before "is":
```go
fmt.Println(year, "is aleap year.")
```